controllers/watcherapi_controller.go

```diff
@@ -188,6 +195,32 @@ func (r *WatcherAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	instance.Status.Conditions.MarkTrue(condition.ServiceConfigReadyCondition, condition.ServiceConfigReadyMessage)
 
+	Log.Info(fmt.Sprintf("[API] Getting input hash '%s'", instance.Name))
+	//
+	// create hash over all the different input resources to identify if any those changed
+	// and a restart/recreate is required.
+	//
+
+	inputHash, hashChanged, errorHash := r.createHashOfInputHashes(ctx, instance, configVars)
+	if errorHash != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ServiceConfigReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.ServiceConfigReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	} else if hashChanged {
+		// Hash changed and instance status should be updated (which will be done by main defer func),
+		// so we need to return and reconcile again
+		return ctrl.Result{}, nil
+	}
```

Comment:
I think it would be appropriate to move L196 (`instance.Status.Conditions.MarkTrue(condition.ServiceConfigReadyCondition, condition.ServiceConfigReadyMessage)`) below this block of code, since only then would the hash be ready and thus `condition.ServiceConfigReadyCondition` should be set to true.

Reply:
that makes sense, done, thanks!